### DISCUSSION TITLE
Platformio integration help

### DISCUSSION THREAD

**Discussion by harry-finch** [SpenceKonde/megaTinyCore#615] [Future Development Discussion]
Hi,

I'd like to add the Atmel SAMD microcontrollers to platformio and was wondering if you could help me with the process since you have done it with MegaTinyCore. Could you give me a general idea of what's needed and maybe documentation if you have some?

I would like to add this core: https://github.com/qbolsee/ArduinoCore-fab-sam

Many thanks!
Harry

**Comment by SpenceKonde** [comment]
Hell no, I didn't do that! @MCUdude did. I have never installed pio in my
life. See the other discussion topics where I keep asking if anyone could
write something up and PR it for pio related stuff because I don't know
enough to synthesize their posts into a guide.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by MCUdude** [reply]
I don't get why you're not just even a little interested in PlatformIO. Arduino IDE is OK for smaller projects, but it has some major drawbacks and is outright outdated. Wouldn't it be nice if the project itself could hold all the settings in the Tools menu? Wouldn't it be nice with a tool that kept track of which libraries and which versions your project needed? Wouldn't it be nice to "left click -> go to implementation"? Wouldn't it be nice if you could specify your own compiler flags, on a per project basis? The list goes on and on.

**Reply by harry-finch** [reply]
Hi @MCUdude 

Do you have some pointers to help me with integrating the core I mentioned above in Platformio? I'd love to do it but I don't really know how to start. Any help would be much appreciated!

**Reply by MCUdude** [reply]
@harry-finch Have a look at [this PR](https://github.com/platformio/platform-atmelmegaavr/pull/18/files). It's mostly just manifest files that sets all the default parameters for the target. The rest is done by the PlatformIO team.

**Reply by apws** [reply]
@harry-finch pls, OT here, but don't you consider also PIC32CM (Cortex-M0+), very similar to ATSAMC20 it seems? (they even advertise it now as cheaper 5V alternative for ATSAMC20) - are they really similar or only rebranded thing??
https://www.microchip.com/en-us/solutions/motor-control-and-drive/motor-control-products/32-bit-microcontrollers-for-motor-control-applications/pic32cm-mc

**Reply by SpenceKonde** [reply]
> I don't get why you're not just even a little interested in PlatformIO. Arduino IDE is OK for smaller projects, but it has some major drawbacks and is outright outdated. Wouldn't it be nice if the project itself could hold all the settings in the Tools menu? Wouldn't it be nice with a tool that kept track of which libraries and which versions your project needed? Wouldn't it be nice to "left click -> go to implementation"? Wouldn't it be nice if you could specify your own compiler flags, on a per project basis? The list goes on and on.

Sure, but I haven't developed any particularly complicated sketches such that I feel the constraints of the IDE, and the core is developed as straight text, in sublime. Hence, there is no particular impetus for me try other development environments. Meanwhile, time spent installing and learning new development environments is time I am not putting towards work that needs to be done, some of which I would like to have complketed by the end of 2020, and am still not through the things that are more urgent than that. Some day I may try it, but I have months upon months of other tasks piled up that are far more important than learning a new IDE when I am not feelling limited by my current tools, I need to set up an entire website, write up two long-form description pages for various initiaatives in order to launch them, I've got >$1000 in inventory that I can't sell because I haven't written up thedescriptions, and I've been promiaing users of my most popular core remedies for all manner of things, but 2.0.0 of ATTinyCore is still not out. Even having left my job, I atill have an incredibly long to-do list, and now things wire direct monetary payback of course take a higher priority. So like... yeah. In a perfect world where all three cores were up to date, all my product listings in Tindie were written up and had photos and I'd finished all my tasks from clients (they take priority above almost everything) and had finished my back taxes (I can't fill them out because I need to dig through years of records, they want to know my "revenue" and "expenses" as if I'm supposed to know what those were!)... If al those things were done - sure I'd love o play with PIO - I might have time in that case to work on projects that were complex enough the benefit from it to. But "learn a new tool which is not of critical importance"? That would be at the bottom of a list wtih hundreds of entries. At least the list isn't getting longer anymore....
